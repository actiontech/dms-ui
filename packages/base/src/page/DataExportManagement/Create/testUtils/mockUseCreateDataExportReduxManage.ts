import { CreateDataExportPageEnum } from '../../../../store/dataExport';
import { ExportMethodEnum } from '../components/CreateTask/ExportMethodForm/index.enum';
import * as useCreateDataExportReduxManage from '../hooks/index.redux';

export const mockCreateDataExportRedux = {
  updateDataExportInfoOpen: false,
  formValues: {
    baseValues: { workflow_subject: 'test', desc: 'desc' },
    sourceValues: { dbService: '121' },
    methodValues: { sql: 'select 1;', exportMethod: ExportMethodEnum.sql }
  },
  pageState: CreateDataExportPageEnum.CREATE_TASK,
  auditLoading: false,
  submitLoading: false,
  taskIDs: ['11111'],
  workflowID: '123',
  updateFormValues: jest.fn(),
  updatePageState: jest.fn(),
  updateAuditLoading: jest.fn(),
  updateSubmitLoading: jest.fn(),
  initModalStatus: jest.fn(),
  updateModalStatus: jest.fn(),
  updateTaskIDs: jest.fn(),
  updateWorkflowID: jest.fn(),
  clearAllState: jest.fn()
};

export const mockUseCreateDataExportReduxManage = (
  data?: Partial<typeof mockCreateDataExportRedux>
) => {
  const spy = jest.spyOn(useCreateDataExportReduxManage, 'default');
  spy.mockImplementation(() => ({ ...mockCreateDataExportRedux, ...data }));
  return spy;
};
