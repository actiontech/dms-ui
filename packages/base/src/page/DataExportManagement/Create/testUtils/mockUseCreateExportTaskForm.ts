import * as useCreateExportTaskForm from '../hooks/useCreateExportTaskForm';

export const mockUseCreateExportTaskFormReturn = {
  baseForm: null as any,
  sourceForm: null as any,
  methodForm: null as any,
  auditLoading: false,
  formatSQLAction: jest.fn(),
  auditAction: jest.fn(),
  resetAllForms: jest.fn()
};

export const mockUseCreateExportTaskForm = (
  data?: Partial<typeof mockUseCreateExportTaskFormReturn>
) => {
  const spy = jest.spyOn(useCreateExportTaskForm, 'default');
  spy.mockImplementation(() => ({
    ...mockUseCreateExportTaskFormReturn,
    ...data
  }));
  return spy;
};
