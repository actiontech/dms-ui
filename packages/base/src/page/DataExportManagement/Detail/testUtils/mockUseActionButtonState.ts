import * as useActionButtonState from '../components/PageHeaderAction/useActionButtonState';

export const mockActionButtonStateData = {
  closeWorkflowButtonMeta: {
    action: jest.fn(),
    loading: false,
    hidden: false
  },
  approveWorkflowButtonMeta: {
    action: jest.fn(),
    loading: false,
    hidden: false
  },
  rejectWorkflowButtonMeta: {
    action: jest.fn(),
    loading: false,
    hidden: false
  },
  executeExportButtonMeta: {
    action: jest.fn(),
    loading: false,
    hidden: false
  }
};

export const mockUseActionButtonState = (
  data?: Partial<typeof mockActionButtonStateData>
) => {
  const spy = jest.spyOn(useActionButtonState, 'default');
  spy.mockImplementation(() => ({ ...mockActionButtonStateData, ...data }));
  return spy;
};
