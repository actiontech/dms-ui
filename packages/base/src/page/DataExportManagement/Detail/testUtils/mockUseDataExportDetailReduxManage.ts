import { GetDataExportTaskStatusEnum } from '@actiontech/shared/lib/api/base/service/common.enum';
import * as useDataExportDetailReduxManage from '../hooks/index.redux';
import { GetDataExportWorkflowResponseData } from '../../../../testUtils/mockApi/dataExport/data';

export const mockDataExportDetailRedux = {
  workflowStepOpen: false,
  workflowRejectOpen: false,
  workflowInfo: GetDataExportWorkflowResponseData,
  taskInfos: [
    {
      task_uid: '1752172791873933312',
      db_info: {
        uid: '1752170583749038080',
        name: 'mysql-1',
        db_type: '',
        database_name: 'test'
      },
      status: GetDataExportTaskStatusEnum.finish,
      export_start_time: '2024-01-30T11:32:15.085+08:00',
      export_end_time: '2024-01-30T11:32:15.085+08:00',
      file_name: 'mysql-1_20240130113114-1752172791873933312.zip',
      audit_result: {
        audit_level: '',
        score: 100,
        pass_rate: 1
      },
      export_type: 'SQL',
      export_file_type: 'CSV'
    },
    {
      task_uid: '1752172791873933313',
      db_info: {
        uid: '1752170583749038080',
        name: 'mysql-1',
        db_type: '',
        database_name: 'test'
      },
      status: GetDataExportTaskStatusEnum.exporting,
      export_start_time: '2024-01-30T11:32:15.085+08:00',
      export_end_time: '2024-01-30T11:32:15.085+08:00',
      file_name: 'mysql-1_20240130113114-1752172791873933313.zip',
      audit_result: {
        audit_level: '',
        score: 100,
        pass_rate: 1
      },
      export_type: 'SQL',
      export_file_type: 'CSV'
    },
    {
      task_uid: '1752172791873933314',
      db_info: {
        uid: '1752170583749038080',
        name: 'mysql-1',
        db_type: '',
        database_name: 'test'
      },
      status: GetDataExportTaskStatusEnum.failed,
      export_start_time: '2024-01-30T11:32:15.085+08:00',
      export_end_time: '2024-01-30T11:32:15.085+08:00',
      file_name: 'mysql-1_20240130113114-1752172791873933314.zip',
      audit_result: {
        audit_level: '',
        score: 100,
        pass_rate: 1
      },
      export_type: 'SQL',
      export_file_type: 'CSV'
    }
  ],
  curTaskID: '1752172791873933312',
  taskStatusNumber: {
    success: 1,
    failed: 0,
    exporting: 0
  },
  canRejectWorkflow: false,
  updateWorkflowStepOpen: jest.fn(),
  updateWorkflowRejectOpen: jest.fn(),
  updateWorkflowInfo: jest.fn(),
  updateTaskInfos: jest.fn(),
  updateCurTaskID: jest.fn(),
  updateTaskStatusNumber: jest.fn(),
  updateCanRejectWorkflow: jest.fn(),
  clearAllDetailState: jest.fn()
};

export const mockUseDataExportDetailReduxManage = (
  data?: Partial<typeof mockDataExportDetailRedux>
) => {
  const spy = jest.spyOn(useDataExportDetailReduxManage, 'default');
  spy.mockImplementation(() => ({ ...mockDataExportDetailRedux, ...data }));
  return spy;
};
