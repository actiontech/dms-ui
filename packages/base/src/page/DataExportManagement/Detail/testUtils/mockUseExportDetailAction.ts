import * as useExportDetailAction from '../hooks/useExportDetailAction';

export const mockExportDetailActionData = {
  refreshWorkflow: jest.fn(),
  closeWorkflowLoading: false,
  closeWorkflow: jest.fn(),
  approveWorkflowLoading: false,
  approveWorkflow: jest.fn(),
  executeExportLoading: false,
  executeExport: jest.fn()
};

export const mockUseExportDetailAction = (
  data?: Partial<typeof mockExportDetailActionData>
) => {
  const spy = jest.spyOn(useExportDetailAction, 'default');
  spy.mockImplementation(() => ({ ...mockExportDetailActionData, ...data }));
  return spy;
};
