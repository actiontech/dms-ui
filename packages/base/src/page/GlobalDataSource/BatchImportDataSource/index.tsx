import {
  BasicButton,
  PageHeader,
  EmptyBox,
  BasicResult
} from '@actiontech/dms-kit';
import { BackButton } from '@actiontech/shared';
import { useTranslation } from 'react-i18next';
import BatchImportDataSourceForm from '../../Project/BatchImportDataSource/UploadForm';
import { ResponseCode } from '@actiontech/dms-kit';
import useBatchImportDataSource from '../../Project/BatchImportDataSource/hooks/useBatchImportDataSource';
import { UploadProps } from 'antd';
import { useCallback } from 'react';
import { LeftArrowOutlined } from '@actiontech/icons';
import { DmsApi } from '@actiontech/shared/lib/api';
import ConnectableErrorModal from '../../Project/BatchImportDataSource/ConnectableErrorModal';
import useBatchCheckConnectable from '../../Project/BatchImportDataSource/hooks/useBatchCheckConnectable';
const GlobalBatchImportDataSource = () => {
  const { t } = useTranslation();
  const {
    importLoading,
    setImportPending,
    setImportDone,
    resultVisible,
    showResult,
    form,
    resetAndHideResult,
    dbServices,
    importServicesCheck,
    setDBservices,
    uploadCheckStatus,
    clearUploadCheckStatus
  } = useBatchImportDataSource();
  const {
    batchCheckConnectable,
    batchCheckConnectableLoading,
    connectErrorModalVisible,
    showConnectErrorModal,
    hideConnectErrorModal,
    connectableInfo
  } = useBatchCheckConnectable();
  const onSubmit = async () => {
    setImportPending();
    DmsApi.ProjectService.ImportDBServicesOfProjectsV2({
      db_services: dbServices
    })
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          showResult();
        }
      })
      .finally(() => {
        setImportDone();
        hideConnectErrorModal();
      });
  };
  const onCheckConnectableBeforeSubmit = async () => {
    await form.validateFields();
    await batchCheckConnectable(dbServices ?? []).then((res) => {
      if (res?.isConnectable) {
        onSubmit();
      } else {
        showConnectErrorModal();
      }
    });
  };
  const onUploadCustomRequest = useCallback<
    Required<UploadProps>['customRequest']
  >(
    (option) => {
      setDBservices([]);
      clearUploadCheckStatus();
      DmsApi.ProjectService.ImportDBServicesOfProjectsCheckV2(
        {
          db_services_file: option.file
        },
        {
          responseType: 'blob'
        }
      )
        .then((res) => {
          importServicesCheck(res);
          option?.onSuccess?.(option?.file);
        })
        .catch((error) => {
          option?.onError?.(error);
        });
    },
    [importServicesCheck, setDBservices, clearUploadCheckStatus]
  );
  return (
    <>
      <PageHeader
        title={
          <BackButton icon={<LeftArrowOutlined />}>
            {t('dmsGlobalDataSource.backToList')}
          </BackButton>
        }
        extra={
          <EmptyBox if={!resultVisible}>
            <BasicButton
              type="primary"
              onClick={onCheckConnectableBeforeSubmit}
              loading={importLoading || batchCheckConnectableLoading}
              disabled={!dbServices?.length}
            >
              {t('dmsGlobalDataSource.batchImportDataSource.submitButton')}
            </BasicButton>
          </EmptyBox>
        }
      />
      <EmptyBox
        if={!resultVisible}
        defaultNode={
          <BasicResult
            status="success"
            title={t('dmsGlobalDataSource.batchImportDataSource.successTitle')}
            extra={[
              <BasicButton
                type="primary"
                key="resetAndClose"
                onClick={resetAndHideResult}
              >
                {t('common.resetAndClose')}
              </BasicButton>
            ]}
          />
        }
      >
        <BatchImportDataSourceForm
          form={form}
          customRequest={onUploadCustomRequest}
          uploadCheckStatus={uploadCheckStatus}
          clearUploadCheckStatus={clearUploadCheckStatus}
        />
      </EmptyBox>
      <ConnectableErrorModal
        modalOpen={connectErrorModalVisible}
        closeModal={hideConnectErrorModal}
        onSubmit={onSubmit}
        loading={importLoading}
        connectErrorList={connectableInfo?.connectErrorList}
      />
    </>
  );
};
export default GlobalBatchImportDataSource;
