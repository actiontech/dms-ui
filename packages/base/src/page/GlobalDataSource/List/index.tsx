import { useEffect, useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import { useNavigate } from 'react-router-dom';
import { message, Modal } from 'antd';
import { TestConnectDisableReasonStyleWrapper } from '@actiontech/shared/lib/components/TestDatabaseConnectButton/style';
import { useDbServiceDriver } from '@actiontech/shared/lib/global';
import { useRequest } from 'ahooks';
import DBService from '@actiontech/shared/lib/api/base/service/DBService';
import { ResponseCode } from '@actiontech/shared/lib/enum';
import {
  ActiontechTable,
  useTableRequestError,
  useTableRequestParams,
  useTableFilterContainer,
  FilterCustomProps,
  TableToolbar,
  TableFilterContainer
} from '@actiontech/shared/lib/components/ActiontechTable';
import { IListGlobalDBService } from '@actiontech/shared/lib/api/base/service/common';
import {
  GlobalDataSourceColumns,
  GLobalDataSourceListParamType
} from './columns';
import eventEmitter from '../../../utils/EventEmitter';
import EmitterKey from '../../../data/EmitterKey';
import useProjectTips from '../../../hooks/useProjectTips';
import useGlobalDataSourceType from '../hooks/useGlobalDataSourceType';
import usePermission from '@actiontech/shared/lib/global/usePermission/usePermission';
import { GlobalDataSourceListActions } from './action';

const GlobalDataSourceList = () => {
  const { t } = useTranslation();

  const navigate = useNavigate();

  const { parse2TableActionPermissions } = usePermission();

  const [modalApi, modalContextHolder] = Modal.useModal();

  const [messageApi, messageContextHolder] = message.useMessage();

  const { getLogoUrlByDbType, updateDriverList } = useDbServiceDriver();

  const {
    updateDbTypeList,
    dbTypeOptions,
    loading: getDbTypeListLoading
  } = useGlobalDataSourceType();

  const {
    projectIDOptions,
    updateProjects,
    loading: getProjectsLoading
  } = useProjectTips();

  const {
    tableFilterInfo,
    updateTableFilterInfo,
    tableChange,
    pagination,
    refreshBySearchKeyword,
    searchKeyword,
    setSearchKeyword
  } = useTableRequestParams<
    IListGlobalDBService,
    GLobalDataSourceListParamType
  >();

  const { requestErrorMessage, handleTableRequestError } =
    useTableRequestError();

  const {
    data: dataSourceList,
    loading,
    refresh
  } = useRequest(
    () => {
      return handleTableRequestError(
        DBService.ListGlobalDBServices({
          ...tableFilterInfo,
          page_index: pagination.page_index,
          page_size: pagination.page_size,
          fuzzy_keyword: searchKeyword
        })
      );
    },
    {
      refreshDeps: [pagination, tableFilterInfo]
    }
  );

  const columns = useMemo(
    () => GlobalDataSourceColumns(getLogoUrlByDbType),
    [getLogoUrlByDbType]
  );

  const { filterButtonMeta, filterContainerMeta, updateAllSelectedFilterItem } =
    useTableFilterContainer(columns, updateTableFilterInfo);

  const filterCustomProps = useMemo(() => {
    return new Map<keyof IListGlobalDBService, FilterCustomProps>([
      ['db_type', { options: dbTypeOptions, loading: getDbTypeListLoading }],
      [
        'project_name',
        { options: projectIDOptions, loading: getProjectsLoading }
      ]
    ]);
  }, [
    dbTypeOptions,
    getDbTypeListLoading,
    projectIDOptions,
    getProjectsLoading
  ]);

  const actions = useMemo(() => {
    const navigateToUpdatePage = (dbServiceUid: string, projectID: string) => {
      navigate(`/project/${projectID}/db-services/update/${dbServiceUid}`);
    };

    const deleteDatabase = (
      dbServiceUid: string,
      dvServiceName: string,
      projectID: string
    ) => {
      const hideLoading = messageApi.loading(
        t('dmsGlobalDataSource.deleteDatabase.deletingDatabase', {
          name: dvServiceName
        }),
        0
      );
      DBService.DelDBService({
        db_service_uid: dbServiceUid,
        project_uid: projectID
      })
        .then((res) => {
          if (res.data.code === ResponseCode.SUCCESS) {
            messageApi.success(
              t('dmsGlobalDataSource.deleteDatabase.deleteSuccessTips', {
                name: dvServiceName
              })
            );
            refresh();
          }
        })
        .finally(() => {
          hideLoading();
        });
    };

    const testDatabaseConnection = (
      dbServiceUid: string,
      dbServiceName: string,
      projectID: string
    ) => {
      const hide = messageApi.loading(
        t('common.testDatabaseConnectButton.testing'),
        0
      );
      DBService.CheckDBServiceIsConnectableById({
        db_service_uid: dbServiceUid,
        project_uid: projectID
      })
        .then((res) => {
          hide();
          if (res.data.code === ResponseCode.SUCCESS) {
            const connections = res.data.data ?? [];
            const isConnectable = connections.every(
              (connection) => !!connection?.is_connectable
            );
            const connectErrorMessage = connections.reduce(
              (acc, cur, curIndex) =>
                !!cur?.is_connectable
                  ? acc
                  : acc +
                    `${cur.component}: ${cur?.connect_error_message?.replace(
                      /\n$/,
                      ''
                    )} ${curIndex < connections.length - 1 ? '\n\r' : ''}`,
              ''
            );
            if (isConnectable) {
              messageApi.success(
                t('common.testDatabaseConnectButton.testSuccess')
              );
            } else {
              modalApi.error({
                title: t('dmsGlobalDataSource.testConnectModal.errorTitle', {
                  dbServiceName
                }),
                content: (
                  <TestConnectDisableReasonStyleWrapper>
                    {connectErrorMessage ?? t('common.unknownError')}
                  </TestConnectDisableReasonStyleWrapper>
                )
              });
            }
          }
        })
        .finally(() => {
          hide();
        });
    };
    return parse2TableActionPermissions(
      GlobalDataSourceListActions(
        navigateToUpdatePage,
        deleteDatabase,
        testDatabaseConnection
      )
    );
  }, [
    parse2TableActionPermissions,
    navigate,
    messageApi,
    t,
    refresh,
    modalApi
  ]);

  useEffect(() => {
    updateDriverList();
    updateProjects();
    updateDbTypeList();
  }, [updateDbTypeList, updateDriverList, updateProjects]);

  useEffect(() => {
    const { unsubscribe } = eventEmitter.subscribe(
      EmitterKey.DMS_Refresh_Global_Data_Source,
      refresh
    );

    return () => {
      unsubscribe();
    };
  }, [refresh]);

  return (
    <>
      {modalContextHolder}
      {messageContextHolder}
      <TableToolbar
        filterButton={{
          filterButtonMeta,
          updateAllSelectedFilterItem
        }}
        searchInput={{
          onChange: setSearchKeyword,
          defaultValue: searchKeyword,
          onSearch: () => {
            refreshBySearchKeyword();
          }
        }}
      />
      <TableFilterContainer
        filterContainerMeta={filterContainerMeta}
        updateTableFilterInfo={updateTableFilterInfo}
        disabled={loading}
        filterCustomProps={filterCustomProps}
      />
      <ActiontechTable
        dataSource={dataSourceList?.list}
        rowKey={(record: IListGlobalDBService) => {
          return `${record?.uid}`;
        }}
        pagination={{
          total: dataSourceList?.total ?? 0,
          current: pagination.page_index
        }}
        loading={loading}
        columns={columns}
        actions={actions}
        errorMessage={requestErrorMessage}
        onChange={tableChange}
      />
    </>
  );
};

export default GlobalDataSourceList;
