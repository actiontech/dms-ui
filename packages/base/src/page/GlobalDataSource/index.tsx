import { Typography } from 'antd';
import { useTranslation } from 'react-i18next';
import { EnterpriseFeatureDisplay } from '@actiontech/shared';
import { PageHeader } from '@actiontech/shared';
import { Outlet } from 'react-router-dom';

const GlobalDataSource = () => {
  const { t } = useTranslation();

  return (
    <section>
      {/* #if [ce] */}
      <PageHeader title={t('dmsGlobalDataSource.pageTitle')} />
      {/* #endif */}
      <EnterpriseFeatureDisplay
        featureName={t('dmsGlobalDataSource.pageTitle')}
        eeFeatureDescription={
          <Typography.Paragraph className="paragraph">
            {t('dmsGlobalDataSource.ceTips')}
          </Typography.Paragraph>
        }
      >
        <Outlet />
      </EnterpriseFeatureDisplay>
    </section>
  );
};

export default GlobalDataSource;
