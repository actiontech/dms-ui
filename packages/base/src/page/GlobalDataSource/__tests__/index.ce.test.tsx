/**
 * @test_version ce
 */

import GlobalDataSource from '..';
import { baseSuperRender } from '../../../testUtils/superRender';

describe('test base/GlobalDataSource ce', () => {
  it('should match snapshot', () => {
    const { container } = baseSuperRender(<GlobalDataSource />);
    expect(container).toMatchSnapshot();
  });
});
