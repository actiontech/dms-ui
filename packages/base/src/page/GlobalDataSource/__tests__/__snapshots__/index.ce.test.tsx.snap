// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test base/GlobalDataSource ce should match snapshot 1`] = `
<div>
  <section />
</div>
`;
